hr_holidays_accrual/wizard/hr_leave_allocation_accrual_calculator.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+
+from odoo import api, fields, models
+
+
+class HrLeaveAllocationAccrualCalculatorAccruement(models.TransientModel):
+    _name = 'hr.leave.allocation.accrual.calculator.accruement'
+    _description = 'HR Leave Allocation Accrual Calculator Accruement'
+
+    calculator_id = fields.Many2one(
+        string='Calculator',
+        comodel_name='hr.leave.allocation.accrual.calculator',
+    )
+    days_accrued = fields.Float(
+        'Number of Days',
+        readonly=True,
+        required=True,
+    )
+    accrued_on = fields.Date(
+        'Accruement Date',
+        readonly=True,
+        required=True,
+    )
+    reason = fields.Char(
+        'Reason',
+        readonly=True,
+        required=True,
+    )
+
+
+class HrLeaveAllocationAccrualBalanceCalculator(models.TransientModel):
+    _name = 'hr.leave.allocation.accrual.calculator'
+    _description = 'HR Leave Allocation Accrual Calculator'
+
+    date = fields.Date(
+        'Date',
+        required=True,
+    )
+    accruement_ids = fields.One2many(
+        string='Accruements',
+        comodel_name=(
+            'hr.leave.allocation.accrual.calculator.accruement'
+        ),
+        inverse_name='calculator_id',
+        readonly=True,
+    )
+    balance = fields.Float(
+        'Balance',
+        readonly=True,
+    )
+
+    @api.multi
+    @api.onchange('date')
+    def _onchange_date(self):
+        self._recalculate()
+
+    @api.multi
+    def _recalculate(self):
+        if not self.date:
+            self.update({
```

Comment:
use loop when you deal with @api.multi

Reply:
Thanks for pointing it out! Seems to me an `ensure_one` should be here